mutiny-core/src/vss.rs

```diff
@@ -0,0 +1,164 @@
+#![allow(dead_code)]
+use crate::auth::MutinyAuthClient;
+use crate::{error::MutinyError, logging::MutinyLogger, storage::MutinyStorage};
+use anyhow::anyhow;
+use lightning::log_error;
+use lightning::util::logger::*;
+use reqwest::{Method, Url};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::sync::Arc;
+
+pub(crate) struct MutinyVssClient<S: MutinyStorage> {
+    auth_client: Arc<MutinyAuthClient<S>>,
+    url: String,
+    logger: Arc<MutinyLogger>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct VssKeyValueItem {
+    pub key: String,
+    pub value: Option<String>,
+    pub version: u32,
+}
+
+impl<S: MutinyStorage> MutinyVssClient<S> {
+    pub(crate) fn new(
+        auth_client: Arc<MutinyAuthClient<S>>,
+        url: String,
+        logger: Arc<MutinyLogger>,
+    ) -> Self {
+        Self {
+            auth_client,
+            url,
+            logger,
+        }
+    }
+
+    pub async fn put_objects(&self, items: Vec<VssKeyValueItem>) -> Result<(), MutinyError> {
+        let url = Url::parse(&format!("{}/putObjects", self.url)).map_err(|e| {
+            log_error!(self.logger, "Error parsing put objects url: {e}");
+            MutinyError::Other(anyhow!("Error parsing put objects url: {e}"))
+        })?;
+
+        // todo do we need global version here?
```

Comment:
Yeah we may need to add something to the auth_client to get the linking_key?

Reply:
this is a global version number, nothing to do with the linking key. I am not really sure the point of the global version number